posixlib/src/main/scala/scala/scalanative/posix/sys/socket.scala

```diff
@@ -9,23 +9,26 @@ object socket {
   type socklen_t   = CUnsignedInt
   type sa_family_t = CUnsignedShort
   type _14         = Nat.Digit2[Nat._1, Nat._4]
-  type sockaddr =
-    CStruct2[sa_family_t,        // sa_family
-             CArray[CChar, _14]] // sa_data, size = 14 in OS X and Linux
+  type sockaddr = CStruct2[sa_family_t,        // sa_family
+             CArray[CChar, _14] // sa_data, size = 14 in OS X and Linux
```

Comment:
Can we somehow force the alignment of these parameter types?

Reply:
Same here. I will go ahead and format now so you can see the whole change. I wrapped a few things and made changes in the PR prior to mass formatting to make sure everything will look good.

Of course, it will be quite a lot of change as explained in this PR but I think it really actually makes readability better.

Reply:
The example above looks like this now.
```scala
  type sockaddr = CStruct2[
    sa_family_t, // sa_family
    CArray[CChar, _14] // sa_data, size = 14 in OS X and Linux
  ]
```
Having the close `]` and or `)` in other place lined up is a key to better readability. I wasn't a believer until I saw it in action.

I am not sure if you are talking about the `=` getting lined up but that is actually a big contributor to whitespace diffs. I think in some cases it increases readability but in others it can be pretty ugly. I was also wanting to more closely match the Scala.js style. Not sure if that is what you were asking.

Reply:
Looks nice! 
Yes, that exactly what I was asking about, alignment of assignments using `=`. I personally prefer them aligned. As you pointed out in the first comment change of this alignment is responsible for most of the diff changes (~23k LOC changed out of ~61k total). However, I've also observed that in Scala.js they're not aligned. If we want to create a common code style across both projects we can follow that convention. 
@sjrd what do you about that?

Reply:
Thanks. I thought I liked that too but wanted to go with the Scala.js `=` non alignment and there are some benefits too like I mentioned, reduced diffs etc. I also wanted to try and go with less configuration. I spent a ton of time trying to upgrade `scalafmt` and never got the result that anyone could buy. Spending a ton of time trying to upgrade or tweak the config to get the perfect format is a losing battle and kind of goes against the philosophy of a formatter. You still sometimes have to preformat or change things a bit to get `scalafmt` to do what you want. I wish that the default was good enough but we are down to very few extra options which is good.

Yes, I think that Scala.js is not forcing alignment of the closing `]` or `)` on a new line but when you look the direction that Scala 3 is going with optional braces and such, I think this is the direction we should go. Because the indents after are always readable, I think this is a good compromise and I think it is actually more readable. Things that fit in 80 chars stay the same, just wrapping is different. Looking at the Scala.js style guide we would just have the closing `)` always on a new line which is different. For example, the `reportError` would have the `)` on a new line. https://github.com/scala-js/scala-js/blob/master/CODINGSTYLE.md#long-expressions-with-binary-operators

Some of the more subtle rules in Scala.js won't be possible for `scalafmt` and certainly using `scalafmt` has some advantages and disadvantages as well. I also think we tend to get used to things over time. I basically had two goals. 1) make the format very similar to the Scala.js guide. 2) Have the smallest configuration file as possible. I also didn't want the diff to be completely out of control. It is big, but I think in a good way.